src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -666,13 +680,19 @@ private async ValueTask<Stream> ConnectToTcpHostAsync(string host, int port, Htt
                     {
                         if (async)
                         {
-                            await socket.ConnectAsync(endPoint, cancellationToken).ConfigureAwait(false);
+                            IPAddress[] addresses = await Dns.GetHostAddressesAsync(host, cancellationToken).ConfigureAwait(false);
+                            activity?.AddEvent(new ActivityEvent("Dns.Resolved"));
+                            await socket.ConnectAsync(addresses, port, cancellationToken).ConfigureAwait(false);
+                            activity?.AddEvent(new ActivityEvent("Transport.Connected"));
```

Comment:
The metric that shows how long it takes to establish a connection and a span that shows an establishment attempt is extremely valuable, while event only allow to visually see it on a trace.

Query that would calculate duration between parent activity start time to event timestamp  would be not a great experience.